### PR TITLE
several documentation updates

### LINKED ISSUE #26: Update help for push command
The push command does not show anything regarding syntax to push different media types, for example

```
oras push localhost:5000/hello:latest hi.txt:application/vnd.me.hi bye.txt:application/vnd.me.bye
```

instead:
```
Push files to remote registry

Usage:
  oras push name[:tag|@digest] file [file...] [flags]
```

### LINKED ISSUE #27: Add documentation on setting up local registry
User should be able to quickly get started with oras using registry running at `localhost:5000`

### LINKED ISSUE #28: Rename workplace to workspace in Docker image
"workspace" is the term more commonly used

### PR DESCRIPTION
**Pull request by jdolitsky**
Several updates to docs, help command, etc. Prepared for a 0.3.1 release w these changes

Fixes #26
Fixes #27
Fixes #28

### CODE REVIEW

File: cmd/oras/pull.go [hunk 1]
```diff
@@ -1,75 +1,86 @@
...
+	username string
+	password string
+}
+
+func pullCmd() *cobra.Command {
+	var opts pullOptions
+	cmd := &cobra.Command{
+		Use:   "pull name[:tag|@digest]",
+		Short: "Pull files from remote registry",
+		Long: `Pull files from remote registry
```
Comment (jdolitsky) [inline cmd/oras/pull.go]: The `Long` fields are the only changes I made. Big diff must be windows vs unix encoding

File: Makefile [hunk 2]
```diff
@@ -19,3 +22,21 @@ covhtml:
...
+
+.PHONY: build-mac
+build-mac:
+	GOARCH=amd64 CGO_ENABLED=0 GOOS=darwin go build -v --ldflags="-w" \
+		-o bin/darwin/amd64/$(CLI_EXE) $(CLI_PKG)
+
+.PHONY: build-windows
+build-windows:
+	GOARCH=amd64 CGO_ENABLED=0 GOOS=windows go build -v --ldflags="-w" \
+		-o bin/windows/amd64/$(CLI_EXE) $(CLI_PKG)
```
Comment (shizhMSFT) [inline Makefile]: For windows, it makes more sense for an executable having a `.exe` extension

Comment (shizhMSFT) [reply Makefile]: It can be in another PR.

File: cmd/oras/pull.go (hunk 1 above)
Comment (shizhMSFT) [reply cmd/oras/pull.go]: CRLF stuffs... always

**Pull request merged**